### PR TITLE
[clang] constexpr built-in abs function.

### PR DESCRIPTION
**Pull request by c8ef**
According to [P0533R9](https://wg21.link/P0533R9), the C++ standard library functions corresponding to the C macros in `[c.math.abs]` are now `constexpr`.

To implement this feature in libc++, we must make the built-in abs function `constexpr`. This patch adds the implementation of a `constexpr` abs function for the current constant evaluator and the new bytecode interpreter.

It is important to note that in 2's complement systems, the absolute value of the most negative value is out of range. In gcc, it will result in an out-of-range error and will not be evaluated as constants. We follow the same approach here.

### CODE REVIEW

File: clang/test/AST/ByteCode/builtin-functions.cpp [hunk 1]
```diff
@@ -265,6 +265,15 @@ namespace fpclassify {
   char classify_subnorm [__builtin_fpclassify(-1, -1, -1, +1, -1, 1.0e-38f)];
 }
 
+namespace abs {
+static_assert(__builtin_abs(14) == 14, "");
+static_assert(__builtin_labs(14L) == 14L, "");
+static_assert(__builtin_llabs(14LL) == 14LL, "");
+static_assert(__builtin_abs(-14) == 14, "");
+static_assert(__builtin_labs(-0x14L) == 0x14L, "");
+static_assert(__builtin_llabs(-0x141414141414LL) == 0x141414141414LL, "");
```
Comment (tbaederr) [inline clang/test/AST/ByteCode/builtin-functions.cpp:274]:
```suggestion
  static_assert(__builtin_abs(14) == 14, "");
  static_assert(__builtin_labs(14L) == 14L, "");
  static_assert(__builtin_llabs(14LL) == 14LL, "");
  static_assert(__builtin_abs(-14) == 14, "");
  static_assert(__builtin_labs(-0x14L) == 0x14L, "");
  static_assert(__builtin_llabs(-0x141414141414LL) == 0x141414141414LL, "");
```

File: clang/lib/AST/ByteCode/InterpBuiltin.cpp [hunk 2]
```diff
@@ -563,6 +563,19 @@ static bool interp__builtin_fabs(InterpState &S, CodePtr OpPC,
   return true;
 }
 
+static bool interp__builtin_abs(InterpState &S, CodePtr OpPC,
+                                const InterpFrame *Frame, const Function *Func,
+                                const CallExpr *Call) {
+  PrimType ArgT = *S.getContext().classify(Call->getArg(0)->getType());
+  APSInt Val = peekToAPSInt(S.Stk, ArgT);
+  if (Val == APSInt(APSInt::getSignedMinValue(Val.getBitWidth()), false))
```
Comment (tbaederr) [inline clang/lib/AST/ByteCode/InterpBuiltin.cpp:571]:
```suggestion
  if (Val == APSInt(APInt::getSignedMinValue(Val.getBitWidth()), /*IsUnsigned=*/false))
```

Same with the other implementation

**Comment by cor3ntin** [comment]
Please add an entry to `clang/docs/ReleaseNotes.rst`, thanks!

File: clang/test/AST/ByteCode/builtin-functions.cpp [hunk 3]
```diff
@@ -265,6 +265,15 @@ namespace fpclassify {
...
 
+namespace abs {
+  static_assert(__builtin_abs(14) == 14, "");
+  static_assert(__builtin_labs(14L) == 14L, "");
+  static_assert(__builtin_llabs(14LL) == 14LL, "");
+  static_assert(__builtin_abs(-14) == 14, "");
+  static_assert(__builtin_labs(-0x14L) == 0x14L, "");
+  static_assert(__builtin_llabs(-0x141414141414LL) == 0x141414141414LL, "");
+} // namespace abs
+
```
Comment (cor3ntin) [inline clang/test/AST/ByteCode/builtin-functions.cpp:276]: can you add the same error test cases as in the other files (abs4, etc)

**Review by cor3ntin — APPROVED** [review]
lgtm

**Pull request merged**